component/src/ibc/event.rs

```diff
@@ -99,8 +105,8 @@ pub fn connection_open_ack(connection_id: &ConnectionId, connection_end: &Connec
                 connection_end
                     .counterparty()
                     .connection_id()
-                    .unwrap()
-                    .to_string(),
+                    .map(|id| id.to_string())
+                    .unwrap_or_default(),
```

Comment:
What's the implication of using unwrap_or_default here? If the field is missing, is it safe to replace with ""? Should we be pre-screening these in the check methods?

Reply:
We should be (and are) doing validation of these optionals in the IBC state machine. There are situations where counterparty connection id is expected to be None, in these cases the cosmos SDK emits this as the empty string. In fact, the raw protos also encode it as an empty string, but `ibc-rs` converts it to an option (where None when "" and Some(val) otherwise)